prae/tests/unchecked.rs

```diff
@@ -0,0 +1,54 @@
+use prae;
+
+prae::define! {
+    pub Username: String
+    ensure |u| !u.is_empty()
+}
+
+#[test]
+#[cfg(feature = "unchecked-access")]
+fn construction_fails_for_invalid_data_unchecked_succeeds() {
+    use assert_matches::assert_matches;
+    use prae::UncheckedAccess;
+
+    assert_matches!(
+        Username::new("").unwrap_err(),
+        prae::ValidationError { .. }
+    );
+
+    let un = Username::new_unchecked("").unwrap();
+    assert_eq!(un.get(), " user ");
+}
+
+#[test]
+#[cfg(feature = "unchecked-access")]
+fn mutation_fails_for_invalid_data_unchecked_succeeds() {
+    use assert_matches::assert_matches;
+    use prae::UncheckedAccess;
+
+    let mut un = Username::new("user").unwrap();
+    assert_matches!(
+        un.try_mutate(|u| *u = "".to_owned()).unwrap_err(),
```

Comment:
This is more of a matter of taste, but let's not call `.unwrap_err()` in `assert_matches!` since it's more consistent with the rest of the tests, a bit shorter and gives a better error in case the method returns `Ok(...)`

Reply:
Done